src/Elastic.Apm/Helpers/TimeExtensions.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+using System.Globalization;
+
+namespace Elastic.Apm.Helpers
+{
+	internal static class TimeExtensions
+	{
+		internal static string FormatForLog(this DateTime dateTime, bool includeKind = true) =>
+			dateTime.ToString("yyyy-MM-dd HH:mm:ss.fffffff", CultureInfo.InvariantCulture) + (includeKind ? $" {dateTime.Kind.FormatForLog()}" : "");
+
+		internal static string FormatForLog(this DateTimeKind dateTimeKind)
+		{
+			switch (dateTimeKind)
+			{
+				case DateTimeKind.Utc: return "UTC";
+				case DateTimeKind.Local: return "Local";
+				case DateTimeKind.Unspecified: return "Unspecified";
+				default: return $"UNRECOGNIZED {dateTimeKind} ({(int)dateTimeKind} as int)";
+			}
+		}
+
+		internal static string FormatForLog(this DateTimeOffset dateTimeOffset) =>
```

Comment:
This is unused.

Reply:
> Is it good though? 🤔

No, now we don't truncate microseconds neither for timestamp nor for durations :)

Reply:
> This is unused.

10x - Fixed.